src/storage/v2/inmemory/storage.cpp

```diff
@@ -1039,8 +1039,30 @@ utils::BasicResult<StorageManipulationError, void> InMemoryStorage::InMemoryAcce
 
 // NOLINTNEXTLINE(google-default-arguments)
 utils::BasicResult<StorageManipulationError, void> InMemoryStorage::InMemoryAccessor::PeriodicCommit(
-    CommitReplArgs /*reparg*/, DatabaseAccessProtector /*db_acc*/) {
-  throw utils::NotYetImplemented("Periodic commit is not yet implemented on in-memory modes.");
+    CommitReplArgs reparg, DatabaseAccessProtector db_acc) {
+  auto result = Commit(reparg, db_acc);
+
+  if (result.HasError()) {
+    return result;
+  }
+
+  FinalizeTransaction();
+
+  auto original_start_timestamp = !transaction_.original_start_timestamp.has_value()
+                                      ? transaction_.start_timestamp
+                                      : transaction_.original_start_timestamp.value();
```

Comment:
It's more readable if you reverse the order (check for a positive result)

Reply:
https://en.cppreference.com/w/cpp/utility/optional/value_or